### PR TITLE
[New Chart] - Add Istio chart

### PR DESCRIPTION
**Pull request by lachie83**
Add incubator chart for https://istio.io/

I think this should be good to go stable once a few folks test it

After install you should be able to run through the BookInfo demo --
 https://istio.io/docs/samples/bookinfo.html

### CODE REVIEW

**Comment by lachie83** [comment]
cc @viglesiasce @prydonius @seanknox

**Comment by rimusz** [comment]
wow it creates five loadbalancers by default, gets pricy.
I think only `istio-ingress`should be enabled by default

**Comment by rimusz** [comment]
not getting much luck on AWS

**Comment by lachie83** [comment]
@rimusz have you pulled the latest commit on this PR? You should only have a single ingress. I thought the same thing.

**Comment by frankbu** [comment]
@lachie83 sorry if this is a stupid question (I'm new to helm charts), but I don't see how you have disabled any of the five loadbalancers. In `values.yaml`, it looks like istio-ingress and all 4 of the addon services are LoadBalancer type and are all enabled. Am I missing something?

**Comment by lachie83** [comment]
my apologies @frankbu you were right. I have pushed the commit. My bad! Thanks for the heads up

File: incubator/istio/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+## Configuration
+
+The following tables lists the configurable parameters of the Istio chart and their default values.
+
+> **Tip**: You can use the default [values.yaml](values.yaml)
+
+Parameter | Description | Default
+--------- | ----------- | -------
+ |  |  |
+
```
Comment (seanknox) [inline incubator/istio/README.md:91]: Let's add the default values here.

**Comment by seanknox** [comment]
UPDATE: I need to configure RBAC in my cluster. @lachie83 added instructions in the README. Was able to install OK.

---
@lachie83 got this error:

`Error: release cantankerous-molly failed: clusterroles.rbac.authorization.k8s.io "cantankerous-molly-istio-manager" is forbidden: attempt to grant extra privileges: [{[*] [istio.io] [istioconfigs] [] []} {[*] [istio.io] [istioconfigs.istio.io] [] []} {[*] [extensions] [thirdpartyresources] [] []} {[*] [extensions] [thirdpartyresources.extensions] [] []} {[*] [extensions] [ingresses] [] []} {[*] [extensions] [ingresses/status] [] []} {[*] [] [configmaps] [] []} {[*] [] [endpoints] [] []} {[*] [] [pods] [] []} {[*] [] [services] [] []}] user=&{system:serviceaccount:kube-system:default 965f2c98-3994-11e7-b810-000d3a749196 [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] map[]} ownerrules=[{[create] [authorization.k8s.io] [selfsubjectaccessreviews] [] []} {[get] [] [] [] [/version /api /api/* /apis /apis/*]}] ruleResolutionErrors=[]`

File: incubator/istio/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: istio
+data:
```
Comment (seanknox) [inline incubator/istio/templates/configmap.yaml:5]: Talked to Lachie and this is a hard requirement of Istio: it's restricted to one install per namespace.

**Review by seanknox — COMMENTED** [review]
Looks great so far.

File: incubator/istio/README.md (hunk 1 above)
Comment (lachie83) [reply incubator/istio/README.md:91]: Will do

**Comment by unguiculus** [comment]
Great. Need to check this out. Could you add standard labels, please?

**Comment by lachie83** [comment]
@unguiculus Sure. Can you kindly confirm the standard labels?

**Comment by andraxylia** [comment]
@lachie83  Related to the RBAC error, did you recently upgrade your cluster? What version do you have?

File: incubator/istio/templates/addons-promethues-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{{if .Values.addons.prometheus.enabled}}
```
Comment (andraxylia) [inline incubator/istio/templates/addons-promethues-deployment.yaml]: Can you correct the typo in the file name?

**Comment by andraxylia** [comment]
I am not very familiar with how helm works, and I have a general question related to how this will be maintained, since I see the istio yaml files were copied and templated. The current files are subject to change... what is the longer term plan?

File: incubator/istio/templates/addons-promethues-deployment.yaml (hunk 3 above)
Comment (lachie83) [reply incubator/istio/templates/addons-promethues-deployment.yaml]: thanks @andraxylia. Updated

**Comment by viglesiasce** [comment]
@andraxylia 

Charts are generally community supported but if the upstream project would like to contribute to the chart they would send a PR.

**Comment by rimusz** [comment]
Addons should be made as subcharts, they do not belong to the main chart.

**Comment by lachie83** [comment]
@rimusz agreed. In fact most of the apps have upstream charts that could be made into dependancies. We can action this outside the scope of this PR

**Comment by rimusz** [comment]
@lachie83 for the start addons as subcharts would be good, later on we can tackle of using upstream charts as dependencies

File: incubator/istio/README.md (hunk 1 above)
Comment (seanknox) [reply incubator/istio/README.md:91]: I'm OK leaving this out for now and tackling in a subsequent PR, especially if we're planning to use dependencies/subcharts. 👍

**Comment by seanknox** [comment]
LGTM: able to install the chart and all istio services/pods appear to work, per their docs.

**Review by seanknox — APPROVED** [review]

**Pull request merged**